### PR TITLE
machines: Show all possible vm states in filtering section

### LINKED ISSUE #14654: machines: Add state of "paused" for Virtual machine list filtering
Page: Machines

Now we have vm list filtering function, but can only choose state of : running & shutoff
It's better to have "paused" state since it's one of the state for vm.
screenshot-10 66 93 123_9090-2020 09 23-14_09_42

### PR DESCRIPTION
**Pull request by skobyda**
In filtering section, use the same list of VM states we use when
defining a VM state in a store.

Fixes #14654

### CODE REVIEW

**Comment by skobyda** [comment]
@garrett When I was working adding more states to a dropdown used for filtering, I noticed that dropdown may be too long and be cutoff. The problem is that `<div id="app">` does not resize for a dropdown. Do you know if there is any CSS solution to resize elements when dropdown is too large?
Screenshot from 2020-09-24 23-52-39

**Comment by garrett** [comment]
From the screenshot, I can tell that this is the result of two different issues: 
1. The page content isn't expanding to the full content area.
2. The dropdowns used are not the correct ones. They look like restyled PF3 dropdowns. Ideally, use PF4, especially [form select](https://www.patternfly.org/v4/documentation/react/components/formselect), if possible. As "form select" is actually using the native widget, it will always pop up over other elements.

Of course # 2 is not much of an issue if # 1 is fixed.

I'll be happy to fix this first issue next week, when I'm back at work. Please ping me about it if I forget. :wink:

**Comment by garrett** [comment]
Hopefully this doesn't break tests. :wink:

(It's a more proper approach than my initial thought of setting `#app { height: 100% }`.)

**Comment by skobyda** [comment]
Thanks, works fell. Now the filtering section looks like this. I'm also wondering if Shut off and Running, the two most used states, should be at the beginning of dropdown.
Screenshot from 2020-09-29 13-48-52

**Comment by marusak** [comment]
@skobyda how much work it would be to generate this list dynamically from states that currently exist in the UI?
I feel like 'No state' is never gonna happen. Also states like 'Crashed' or 'Shutting down' would be very rare cases (the second one because it is just intermediate state). And as you wrote that you wonder if some states should be on the top of the list - if we would have only states that currently apply there would be no need to think what should be promoted.

**Comment by garrett** [comment]
I agree about having running / shut off at the top, then a divider, then the rest of the states from the list.

If there's only running and shut off states, then drop the divider.

If there's only running or only shut off, it's debatable if we really want a 1 option dropdown. They're both common enough that perhaps we should _always_ have running and shut off states.

We could extend the 2 common states to 3 if we want to include paused in there. But the rest are mainly error states and should come later, certainly.

**Comment by skobyda** [comment]
I updated it so that only states which are present in the VM list are shown and Running and Shut off are present at the beginning of the list. But there is no divider because I cannot find the proper way to use divider in PF4 Selector

File: pkg/machines/hostvmslist.jsx
Comment (marusak) [inline pkg/machines/hostvmslist.jsx:111]: forgotten debuging line

Comment (marusak) [inline pkg/machines/hostvmslist.jsx:111]: forgotten debugging log

**Review by marusak — CHANGES_REQUESTED** [review]
I haven't built this yet, but from the code it looks like if all VMs are in the same state that you still would have dropdown with [All, The_one_state]. That seems pointless. Could we not show the whole filter then? Also when there are no VMs at all.

Also tests seems unhappy, it might likely be related, can you please take a look? Thanks!

**Comment by skobyda** [comment]
I updated it so that only states which are present in the VM list are shown and Running and Shut off are present at the beginning of the list. But there is no divider because I cannot find the proper way to use divider in PF4 Selector. Do you know the proper way?

Screenshot from 2020-10-02 14-39-35

File: pkg/machines/hostvmslist.jsx
Comment (KKoukiou) [inline pkg/machines/hostvmslist.jsx:140]: The Vm states need to be translated.

**Review by KKoukiou — CHANGES_REQUESTED** [review]

**Comment by garrett** [comment]
Is that a PF4 select? It isn't styled like one in the screenshot. Odd.

On the first select demo, there's a divider:
https://www.patternfly.org/v4/components/select/react/

Excerpt (with one line context):

```jsx
...
        <SelectOption key={4} value="Ms" />,
        <Divider component="li" key={5} />,
        <SelectOption key={6} value="Dr" />,
...
```

**Comment by skobyda** [comment]
> Is that a PF4 select? It isn't styled like one in the screenshot. Odd.
> 
> On the first select demo, there's a divider:
> https://www.patternfly.org/v4/components/select/react/
> 
> Excerpt (with one line context):
> 
> ```js-jsx
> ...
>         <SelectOption key={4} value="Ms" />,
>         <Divider component="li" key={5} />,
>         <SelectOption key={6} value="Dr" />,
> ...
> ```

Oh damn, but it has a bug. Try to go here https://www.patternfly.org/v4/components/select/react/,click on first select, select the last option and you will get TypeError: Cannot read property 'toString' of undefined. I get same error when I applied it here.

**Comment by skobyda** [comment]
It seems they are fixing it: https://github.com/patternfly/patternfly-react/pull/4998

**Comment by KKoukiou** [comment]
> > Is that a PF4 select? It isn't styled like one in the screenshot. Odd.
> > On the first select demo, there's a divider:
> > https://www.patternfly.org/v4/components/select/react/
> > Excerpt (with one line context):
> > ```
> > ...
> >         <SelectOption key={4} value="Ms" />,
> >         <Divider component="li" key={5} />,
> >         <SelectOption key={6} value="Dr" />,
> > ...
> > ```
> 
> Oh damn, but it has a bug. Try to go here https://www.patternfly.org/v4/components/select/react/,click on first select, select the last option and you will get TypeError: Cannot read property 'toString' of undefined. I get same error when I applied it here.

This issue has already being reported. https://github.com/patternfly/patternfly-react/issues/4945 
If you feel like fixing it I can help you get around the patternfly repository and testing tools.

**Comment by skobyda** [comment]
> > > Is that a PF4 select? It isn't styled like one in the screenshot. Odd.
> > > On the first select demo, there's a divider:
> > > https://www.patternfly.org/v4/components/select/react/
> > > Excerpt (with one line context):
> > > ```
> > > ...
> > >         <SelectOption key={4} value="Ms" />,
> > >         <Divider component="li" key={5} />,
> > >         <SelectOption key={6} value="Dr" />,
> > > ...
> > > ```
> > 
> > 
> > Oh damn, but it has a bug. Try to go here https://www.patternfly.org/v4/components/select/react/,click on first select, select the last option and you will get TypeError: Cannot read property 'toString' of undefined. I get same error when I applied it here.
> 
> This issue has already being reported. [patternfly/patternfly-react#4945](https://github.com/patternfly/patternfly-react/issues/4945)
> If you feel like fixing it I can help you get around the patternfly repository and testing tools.

But there is already a fix for that. See my comment above.

**Comment by marusak** [comment]
@skobyda Today we got https://github.com/cockpit-project/cockpit/issues/14759 which has very similar problem as you were fixing here. Can you please check if by that fix in this PR you fixed that issue? If so, can you please add Fixes line, if not would it make sense to fix it together for all dropdowns?
Thanks!

p.s. There are conflicts, can you also rebase?
p.s.2 Is this blocked on the aforementioned issue? Can we do something about it to unblock ourselves?

**Review by KKoukiou — APPROVED** [review]
Rebased because it has a merge conflict and force pushed. The PR looks good otherwise.

**Comment by marusak** [comment]
Only testStoragePools is broken but that is unrealated and being fixed in https://github.com/cockpit-project/cockpit/pull/14778/commits/3f0e7ce6b1594ccfee7fa7e69d5110fc66637bb9

**Review by marusak — APPROVED** [review]

**Pull request merged**
